code/missionui/missionbrief.cpp

```diff
@@ -976,6 +976,62 @@ void brief_init()
    Brief_inited = TRUE;
 }
 
+// --------------------------------------------------------------------------------------
+// brief_api_init()
+//
+// A pared down version of brief_init() for use with the Scpui API. Removes initialization of
+// elements not used by the API but still allows drawing of the briefing map - Mjn
+//
+void brief_api_init()
+{
+	// get a pointer to the appropriate briefing structure
+	if (MULTI_TEAM) {
+		Briefing = &Briefings[Net_player->p_info.team];
+	} else {
+		Briefing = &Briefings[0];
+	}
+
+	Brief_last_auto_advance = 0;
+
+	brief_compact_stages(); // compact the briefing array to eliminate unused stages
+
+	// The API handles mission goals on it's own, but we still need to count the stages and add one
+	// if appropriate -Mjn
+	if (The_mission.flags[Mission::Mission_Flags::Toggle_showing_goals] ==
+		!!(The_mission.game_type & MISSION_TYPE_TRAINING)) {
+		Num_brief_stages = Briefing->num_stages + 1;
+	} else {
+		Num_brief_stages = Briefing->num_stages;
+	}
+
+	Current_brief_stage = 0;
+	Last_brief_stage = 0;
+
+	// init the scene-cut data
+	brief_transition_reset();
+
+	//This will need to be replaced with an API version in a later PR that deals with
+	//weapon and ship select APIs - Mjn
+	common_select_init();
+
+	// init the briefing map
+	brief_init_map();
+
+	// set the camera target
+	if (Briefing->num_stages > 0) {
+		brief_set_new_stage(&Briefing->stages[0].camera_pos,
+			&Briefing->stages[0].camera_orient,
+			0,
+			Current_brief_stage);
+		brief_reset_icons(Current_brief_stage);
+	}
+
+	Brief_mouse_up_flag = 0;
+	Closeup_font_height = gr_get_font_height();
+	Closeup_icon = NULL;
```

Comment:
this should be ``nullptr`` instead of ``NULL``.
``NULL`` is a remnant of old C-style code, and while it's not too much of a problem, we intend to replace it in code that we touch as to eventually get rid of all NULLs